File: sdks/ruby/lib/daon/verification_result.rb
module Daon
  class VerificationResult
    attr_reader :verified, :content_hash, :creator, :license, :timestamp, 
                :platform, :verification_url, :error

    def initialize(verified:, content_hash:, creator: nil, license: nil, 
                   timestamp: nil, platform: nil, verification_url: nil, error: nil)
      @verified = verified
      @content_hash = content_hash
      @creator = creator
      @license = license
      @timestamp = timestamp
      @platform = platform
      @verification_url = verification_url
      @error = error
    end

    def protected?
      verified
    end

    def unprotected?
      !verified
    end

    def liberation_licensed?
      license&.include?('liberation')
    end

    def creative_commons?
      license&.start_with?('cc_')
    end

    def all_rights_reserved?
      license == 'all_rights_reserved'
    end

    def protection_date
      timestamp
    end

    def blockchain_url
      return nil unless content_hash
      "https://explorer.daon.network/content/#{content_hash}"
    end

    def to_h
      {
        verified: verified,
        content_hash: content_hash,
        creator: creator,
        license: license,
        timestamp: timestamp&.iso8601,
        platform: platform,
        verification_url: verification_url,
        blockchain_url: blockchain_url,
        error: error
      }
    end

    def to_json
      to_h.to_json
    end
  end
end

File: sdks/php/src/VerificationResult.php
<?php

namespace Daon;

class VerificationResult
{
    private bool $verified;
    private string $contentHash;
    private ?string $license;
    private ?\DateTime $timestamp;
    private ?string $verificationUrl;
    private ?string $blockchainUrl;
    private ?string $error;

    public function __construct(array $data)
    {
        $this->verified = $data['verified'] ?? false;
        $this->contentHash = $data['content_hash'] ?? '';
        $this->license = $data['license'] ?? null;
        $this->timestamp = $data['timestamp'] ?? null;
        $this->verificationUrl = $data['verification_url'] ?? null;
        $this->blockchainUrl = $data['blockchain_url'] ?? null;
        $this->error = $data['error'] ?? null;
    }

    public function isVerified(): bool { return $this->verified; }
    public function getContentHash(): string { return $this->contentHash; }
    public function getLicense(): ?string { return $this->license; }
    public function getTimestamp(): ?\DateTime { return $this->timestamp; }
    public function getVerificationUrl(): ?string { return $this->verificationUrl; }
    public function getBlockchainUrl(): ?string { return $this->blockchainUrl; }
    public function getError(): ?string { return $this->error; }
}
